src/dataServices/dynamoDbDataService.ts

```diff
@@ -94,13 +97,21 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
 
     async createResource(request: CreateResourceRequest) {
         const { resourceType, resource, id } = request;
+
         const vid = 1;
         let item = resource;
         item.resourceType = resourceType;
         item.meta = generateMeta(vid.toString());
 
         const params = DynamoDbParamBuilder.buildPutAvailableItemParam(item, id || uuidv4(), vid);
-        await this.dynamoDb.putItem(params).promise();
+        try {
+            await this.dynamoDb.putItem(params).promise();
+        } catch (e) {
+            if (e.code === 'ConditionalCheckFailedException') {
+                throw new Error(`Unable to create resource with id ${id}, because resource already exists`);

```

Comment:
In a create we have control of the `id`. Instead of an error can we generate a new uuid and retry?

Reply:
If we have an `id` conflict, it's unlikely that the `id` was a randomly generated `uuid`. The conflicting `id` most likely came from the request. If the `id` did come from the request, I think we should let the requester know that the `id` they provided didn't work rather than generate a random new `id` for them.

Reply:
hmmmmm So this will result in a 500 returned 'InternalServiceError'

If the user didn't provide an id we should return a 'Retryable' but if they did provide an id we should  return a 422 Unprocessable Entity or 400 Bad Request

Reply:
Updated